### PR TITLE
Added support to manage Project's groups

### PR DESCRIPTION
**Pull request by ghost**
- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

I exposed the [Project Group API](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-GetProjectsGroups) with this pull request. 

I tested it manually on our production (\o/) Artifactory Cloud instance and it works well.
However, I was not able however to implement test coverage. I ran:
```
go test -v github.com/jfrog/jfrog-client-go/tests -timeout 0 -test.access
```
and got:
```
=== RUN   TestAccessVersion
--- FAIL: TestAccessVersion (0.00s)
panic: Failed: Method is not implemented [recovered]
	panic: Failed: Method is not implemented

goroutine 6 [running]:
testing.tRunner.func1.2({0x99e740, 0xb460e0})
	/usr/local/go/src/testing/testing.go:1209 +0x24e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1212 +0x218
panic({0x99e740, 0xb460e0})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/jfrog/jfrog-client-go/access/auth.(*accessDetails).GetVersion(0x0)
	/home/andrea/repos/jfrog-client-go/access/auth/accessdetails.go:16 +0x27
github.com/jfrog/jfrog-client-go/tests.TestAccessVersion(0xc0001b84e0)
	/home/andrea/repos/jfrog-client-go/tests/access_test.go:10 +0x74
testing.tRunner(0xc0001b84e0, 0xaac7f0)
	/usr/local/go/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1306 +0x35a
FAIL	github.com/jfrog/jfrog-client-go/tests	0.007s
FAIL
```

I also tried to run the existing Test directly:
```
go test -v github.com/jfrog/jfrog-client-go/tests -timeout 0 -test.access -rt.url=http://localhost:8081/ -run TestAccessProject
```
and got:
```
=== RUN   TestAccessProject
=== RUN   TestAccessProject/create-update-delete
[Debug] Sending HTTP GET request to: api/v1/projects/tstprj
[Debug] Sending HTTP GET request to: api/v1/projects/tstprj
[Warn] Attempt 0 - Failure occurred while sending GET request to api/v1/projects/tstprj - Get "api/v1/projects/tstprj": unsupported protocol scheme ""
    accessproject_test.go:19: 
        	Error Trace:	accessproject_test.go:19
        	Error:      	Received unexpected error:
        	            	Get "api/v1/projects/tstprj": unsupported protocol scheme ""
        	Test:       	TestAccessProject/create-update-delete
[Debug] Sending HTTP PUT request to: api/v1/projects/tstprj
[Debug] Sending HTTP PUT request to: api/v1/projects/tstprj
[Warn] Attempt 0 - Failure occurred while sending PUT request to api/v1/projects/tstprj - Put "api/v1/projects/tstprj": unsupported protocol scheme ""
    accessproject_test.go:23: 
        	Error Trace:	accessproject_test.go:23
        	Error:      	Received unexpected error:
        	            	Put "api/v1/projects/tstprj": unsupported protocol scheme ""
        	Test:       	TestAccessProject/create-update-delete
[Debug] Sending HTTP GET request to: api/v1/projects/tstprj
[Debug] Sending HTTP GET request to: api/v1/projects/tstprj
[Warn] Attempt 0 - Failure occurred while sending GET request to api/v1/projects/tstprj - Get "api/v1/projects/tstprj": unsupported protocol scheme ""
    accessproject_test.go:25: 
        	Error Trace:	accessproject_test.go:25
        	Error:      	Received unexpected error:
        	            	Get "api/v1/projects/tstprj": unsupported protocol scheme ""
        	Test:       	TestAccessProject/create-update-delete
[Debug] Sending HTTP DELETE request to: api/v1/projects/tstprj
[Debug] Sending HTTP DELETE request to: api/v1/projects/tstprj
[Warn] Attempt 0 - Failure occurred while sending DELETE request to api/v1/projects/tstprj - Delete "api/v1/projects/tstprj": unsupported protocol scheme ""
    accessproject_test.go:33: 
        	Error Trace:	accessproject_test.go:33
        	            				panic.go:1038
        	            				panic.go:221
        	            				signal_unix.go:735
        	            				accessproject_test.go:26
        	Error:      	Received unexpected error:
        	            	no response provided (including status code)
        	Test:       	TestAccessProject/create-update-delete
--- FAIL: TestAccessProject (0.00s)
    --- FAIL: TestAccessProject/create-update-delete (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8fe9d9]

goroutine 20 [running]:
testing.tRunner.func1.2({0x9c0b20, 0xf5e600})
	/usr/local/go/src/testing/testing.go:1209 +0x24e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1212 +0x218
panic({0x9c0b20, 0xf5e600})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/jfrog/jfrog-client-go/tests.testAccessProjectCreateUpdateDelete(0xc0001c6d00)
	/home/andrea/repos/jfrog-client-go/tests/accessproject_test.go:26 +0x359
testing.tRunner(0xc0001c6d00, 0xaacd40)
	/usr/local/go/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1306 +0x35a
FAIL	github.com/jfrog/jfrog-client-go/tests	0.006s
FAIL
```

I was running it against a local Artifactory Pro docker instance, but I don't have the license to set it up, but not sure if that is the problem as it complains about (http?) scheme:
```
docker run --name artifactory -v /opt/artifactory/var/:/var/opt/jfrog/artifactory -d -p 8081:8081 -p 8082:8082 releases-docker.jfrog.io/jfrog/artifactory-pro:latest
```

Let me know how I can improve on this, thank you!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1126,6 +1130,30 @@ err = accessManager.AssignRepoToProject("repoName", "tstprj", true)
 err = accessManager.AssignRepoToProject("repoName")
 ```
 
+#### Get all groups assigned to a project
+```go
+err = accessManager.GetProjectsGroups("tstprj")
```
Comment (eyalbe4) [inline README.md:1135]: The documented API returned values should be corrected.

Let's rename "tstprj" to "my-project".

File: README.md [hunk 2]
```diff
@@ -1126,6 +1130,30 @@ err = accessManager.AssignRepoToProject("repoName", "tstprj", true)
...
+```
+
+#### Get a specific group assigned to a project
+```go
+err = accessManager.GetProjectsGroup("tstprj", "tstgroup")
```
Comment (eyalbe4) [inline README.md:1140]: The documented API returned values should be corrected.

Let's rename "tstprj" to "my-project".
Let's rename "tstprj" to "my-group".

Let's also rename the sample arguments in the new ```UpdateGroupInProject``` and ```DeleteExistingProjectGroup ``` APIs.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Thanks you for this contribution!
As for the issues you've been gaving with the tests, try implementing the test code, without checking it. We will then assist by running it ourselves, and fixing it if needed.
You can implement the following test logic - 
1. Create a project
2. Add a grouo to the project
3. Get the group from the project and validate it
4. Update the groupand validate it
5. Delete the group ans see that its been deleted
6. Delete the project

**Comment by amazzotti** [comment]
Hello @eyalbe4, I did my best with the test. 
Since I can't run it I'm not sure it will work, but I hope so. 

Let me know if there is anything more I can do. 
Thank you for your support!

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
I believe that the new ```testAccessProjectCreateUpdateDelete``` added in this PR never runs, because it's not invoked by the ```TestAccessProject``` function, like the ```testAccessProjectCreateUpdateDelete``` function is.

Also note that we have rebuilt jfrog-client-go's testing platform, and we can now initiate the tests following every commit to a PR. The tests are initiated once one of the project maintainers adds the "Safe to test" label to the PR. I've just done that, and you can see the tests under *Checks* in the PR's *Conversation* tab.

You'll see that all tests pass, but as I said, you test didn't run.
Please go ahead and fix the test, and we'll rerun the tests. Feel free to add more commits with fixes in case the tests fail.

**Comment by eyalbe4** [comment]
@amazzotti - 
See the ```TestAccessProjectGroups``` run - https://github.com/jfrog/jfrog-client-go/runs/5002650944?check_suite_focus=true

**Comment by amazzotti** [comment]
> @amazzotti - See the `TestAccessProjectGroups` run - https://github.com/jfrog/jfrog-client-go/runs/5002650944?check_suite_focus=true

Yes I saw that and I added a commit that should create (and delete after tests) the group I'm using with access. 

Not the easiest to code blindly, but I like the test label. 

How do I run tests again?

**Comment by eyalbe4** [comment]
Sure @amazzotti. No problem.
We will rerun the tests for you following every commit.

**Comment by amazzotti** [comment]
May you help me out in here? The DeleteGroup call fails but I'm not sure why and I can't debug it. :(

**Comment by eyalbe4** [comment]
Sure @amazzotti -
I think that you should initialize the ```testGroupService``` right after the initialization of ```testsAccessProjectService```
image

**Comment by amazzotti** [comment]
> Sure @amazzotti - I think that you should initialize the `testGroupService` right after the initialization of `testsAccessProjectService` image

Sorry this is a bit convoluted. I used initArtifactoryTest but this is dead code it seems. 
The testGroupService should be initialized somewhere else already, may I reuse that code?

**Comment by amazzotti** [comment]
Have you thought maybe at offering a trial license for Access like other products are doing? I'm thinking about Atlassian as a model here. 

This would really benefit development and contributions.

**Comment by amazzotti** [comment]
Oh and please do run the tests again, I hope it's fixed now.

**Comment by eyalbe4** [comment]
Done @amazzotti.

**Comment by amazzotti** [comment]
Still failing. 
I need some help here or the ability of running these tests locally. 
It's really hard this way.

**Comment by amazzotti** [comment]
I also think the code I added it's pointless. 
As I said before the testGroupService should already be initialized, since the group is correctly created. 
Somehow the Delete method just fails, but I can't debug that.

**Comment by eyalbe4** [comment]
@amazzotti - 
I'll try to put aside some time this week to look into your code, and share with you here my findings.

**Comment by amazzotti** [comment]
> @amazzotti - I'm so sorry for not getting to helping with the test fix. We really appreciate your contribution and would like to see it merged. Here's something we can do so that you're not blocked. You can set a free JFrog instance in the cloud, and use it to debug the tests. To set it up, all you need to do is run this cUrl command -
> 
> ```
> curl -fL https://getcli.jfrog.io/setup | sh
> ```
> 
> Please let me know if you need further assistance with this.

The curl command actually installs the cli tool, which I don't need. 

However I could initialize a free JFrog cloud instance and that comes with the Access functionality that I needed. 
That's great, thank you for pointing that out. 

I was able to run the tests correctly and I fixed a couple of issues, should be all good now. Please run the tests again to confirm.

**Comment by eyalbe4** [comment]
Thanks so much for this great contribution @amazzotti!
v1.9.0 is now released to include this featire.

**Pull request merged**